### PR TITLE
Update user

### PR DESCRIPTION
**Pull request by aqizzz**

### CODE REVIEW

File: server/src/routes/user.route.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Router } from 'express';
+import { createUser, deleteUser, login, updateUser, getUser } from '../controllers';
+import { authenticateJWT } from '../middlewares/jwt.middleware';
+
+export const userRoute = () => {
+	const router = Router();
+
+  router.get('/user/user', authenticateJWT, getUser);
```
Comment (XintangZ) [inline server/src/routes/user.route.ts:8]: /user

File: server/src/routes/user.route.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Router } from 'express';
+import { createUser, deleteUser, login, updateUser, getUser } from '../controllers';
+import { authenticateJWT } from '../middlewares/jwt.middleware';
+
+export const userRoute = () => {
+	const router = Router();
```
Comment (XintangZ) [inline server/src/routes/user.route.ts:6]: extra indentation

File: server/src/controllers/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+		return res.status(200).json({ data: updatedUser });
+	} catch (error) {
+		console.error('Error updating user:', error);
+		return res.status(500).json({ message: 'Server error' });
+	}
+};
+
+const deleteUser = async (req: Request, res: Response) => {
+  const id = res.locals.user.userId;
```
Comment (XintangZ) [inline server/src/controllers/user.controller.ts:137]: inconsistent indentation

File: server/src/routes/user.route.ts
Comment (XintangZ) [inline server/src/routes/user.route.ts:1]: suggested endpoint design:

  // Get user profile (protected route)
        router.get('/users/me', authenticateJWT, getUser);

  // User login (public route)
        router.post('/users/login', login);

  // Create a new user (public route)
        router.post('/users', createUser);

  // Update user profile (protected route)
        router.put('/users/me', authenticateJWT, updateUser);

  // Delete user profile (protected route)
        router.delete('/users/me', authenticateJWT, deleteUser);

**Review by XintangZ — COMMENTED** [review]
code formatting suggestion: there are some inconsistent indentation, consider using prettier for auto formating

**Review by XintangZ — APPROVED** [review]

**Review by Carolyn1204 — APPROVED** [review]

**Pull request merged**